### DISCUSSION TITLE
How do I extend the default user object?

### DISCUSSION THREAD

**Discussion by italodeandra** [nextauthjs/next-auth#3299] [Help]
Hi team,

I'm currently in the need of extending the default user object, and I did it like this:

```tsx
const MongoDBAdapterWrapper: typeof MongoDBAdapter = (options) => {
  const adapter = MongoDBAdapter(options);
  adapter.createUser = (user) => {
    user.type = UserType.NORMAL;
    return adapter.createUser(user);
  };
  return adapter;
};
```

Is there a better way for doing this or did I manage to do in a good way? For future "extensions" should I keep doing the same way?

**Comment by balazsorban44** [comment]
you don't need any of that. 

For TS:

https://next-auth.js.org/getting-started/typescript

Further, the `createUser` forwards anything to your db automatically, that's returned in the `profile` callback of your provider

https://next-auth.js.org/adapters/models#user

**Reply by italodeandra** [reply]
I'm using the `EmailProvider` so there isn't a `profile` callback.

**Reply by joematune** [reply]
@italodeandra - I see what you mean - provider `type = "email"` doesn't have a profile callback to override. It's created [here](https://github.com/nextauthjs/next-auth/blob/50fe115df6379fffe3f24408a1c8271284af660b/src/core/routes/callback.ts#L218) and it might be nice to add to the initial profile as it's currently just an email. _i.e._
```diff
// If it is an existing user, use that, otherwise use a placeholder
const profile = (identifier
  ? await getUserByEmail(identifier)
  : null) ?? {
+ ...initialProfile,
  email: identifier,
}
```

@balazsorban44 Thanks for the comment. I'm using an `oauth` provider so passing in a `profile` callback in the provider options to [override the default](https://next-auth.js.org/configuration/providers/oauth#override-default-options) helped 🙌. Is the above snippet something worth looking into?
